ApiCustomer/Data/CostumerApi/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Data.Entities;
+using Data.Repositories;
+using Data.Validator;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CostumerApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private readonly ICustomerRepository _repository;        
+
+        CustomerValidator _validator = new();
+
+        public CustomerController(ICustomerRepository repository)
+        {
+            _repository = repository;
+        }
+
+        [HttpDelete]
+        public IActionResult Delete(long id)
+        {
+            var result = _repository.Delete(id);
+
+            if (result == "404")
+                return NotFound();
+
+
+            return Ok(result);
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            return Ok(_repository.GetAll());
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetById(long id)
+        {
+            var result = _repository.GetById(id);
+
+            if (result == null)
+                return NotFound();
```

Comment:
Necessário retornar a informação de qual o campo não encontrou resultado.
```suggestion
                return NotFound($"Did not found customer for Id: {id}");
```